frappe/model/rename_doc.py

```diff
@@ -20,9 +20,15 @@ def update_document_title(doctype, docname, title_field=None, old_title=None, ne
 	if docname and new_name and not docname == new_name:
 		docname = rename_doc(doctype=doctype, old=docname, new=new_name, merge=merge)
 
-	if old_title and new_title and not old_title == new_title:
```

Comment:
Why remove this condition?

Reply:
That was unintentional, added it back. Thanks!